sympy/physics/mechanics/kane.py

```diff
@@ -317,6 +317,18 @@ def _initialize_constraint_matrices(self, config, vel, acc, linear_solver='LU'):
                 self._f_dnh = msubs(acc, udot_zero)
                 self._k_dnh = (acc - self._f_dnh).jacobian(self._udot)
 
+            # The linear velocity constraints must be linear in both q'
+            # and u, so check for u and q' in the components.
```

Comment:
q' shouldn't be present in vel at all. Line 302 replaces all q' using the solution to the kinematical differential equations.

Reply:
> q' shouldn't be present in vel at all. Line 302 replaces all q' using the solution to the kinematical differential equations.

I noticed this in my many tests, but did not pay attention. I will change the description.